dtfs-central-api/src/cron-scheduler-jobs/deal-cancellation/cancel-deal-job.ts

```diff
@@ -11,13 +12,7 @@ const { DEAL_CANCELLATION_SCHEDULE } = process.env;
  */
 const cancelDeals = async (deals: TfmDealWithCancellation[]) => {
   await Promise.all(
-    deals.map((deal) =>
-      TfmDealCancellationRepo.submitDealCancellation({
-        dealId: deal._id,
-        cancellation: deal.tfm.cancellation,
-        auditDetails: generateSystemAuditDetails(),
-      }),
-    ),
+    deals.map(async (deal) => DealCancellationService.submitScheduledCancellation(deal._id, deal.tfm.cancellation, generateSystemAuditDetails())),
```

Comment:
Leaving my comment here so you understand my confusion: Just for my understanding are we always submitting a scheudled cancellation even if its instantly actionable?

---

I get this now, but I think with fresh eyes, we could reword some things -- This is a cron job cancelling deals (docs could be `Cancels deals` i think and leave it at that). 

I wouldn't say this is submitting anything? I'd say this is a `cancelDealInTfmAndPortal` would be more accurate? Or `processScheduledCancellation` or even just `processDealCancellation`... I don't know, I get the naming now I've dived but it wasn't clear to me (and I'm a good litmus test for this as I have been hands off).

Bascially if I call a function called `cancelDeals` I'd expect it to cancel a deal, not submit a scheduled cancellation. But I do get it, so if this is the best way I'm happy to keep as is.

(sorry for a trail of thought comment):

I'd probably expect us to have a 
`addDealCancellationAmendment` (or whatever) and a  `processDealCancellation` -- the former always gets called on amendment, the later either gets called on the amendment or in the cron job depending on timing? 

Again, I'm pragmatic about my comments, so feel free to say `no because x,y,z` :)

Reply:
Hmm this needs some thought

Reply:
So currently the service has two methods `submitScheduledCancellation` and `cancelDeal`. I was considering changing the latter as I think its not quite descriptive enough but in the end left it be. I do see your argument about `submit`, these are my thoughts on what the name should be
- mention `scheduled` - it shouldnt be used on a cancellation that hasnt been scheduled as it doesn't send an email nor add an activity
- make it clear the cancellation is being enacted, not the scheduling being cancelled (if that makes sense) - I think a name like `cancelScheduledDeal` could be confused with the `schedule` being cancelled

With that in mind maybe `processScheduledCancellation` is the best choice?

Whilst we're on this train of thought I think `cancelDeal` should be changed to `submitDealCancellation` - since it doesn't always cancel the deal

Thoughts?

Reply:
We had a look at this -- as it's not expressively part of this review we're leaving this for now, but I've had a chat with Mark about ways we can improve in future